### PR TITLE
Graph view improvements

### PR DESCRIPTION
**Pull request by Arne-W**
Hi,
I added an optimized method to the log-storage for getting values. It avoids an additional copy of the data. Moreover all log data that haves strings as values are suppressed in the Graph tree, as it is not useful to plot them. If we get more text messages they should be handled like error, mode,... messages.
Next I updated the copter log formatters to reflect the latest changes in ArduPilot.

### CODE REVIEW

**Comment by Arne-W** [comment]
@billbonney could you please review - next PR is already on the way :smiley:

File: src/ui/AP2DataPlot2D.cpp [hunk 1]
```diff
@@ -1074,7 +1053,7 @@ void AP2DataPlot2D::itemEnabled(QString name)
         }
         // use golden ratio for evenly distributed colors
         double golden_ratio_conjugate = 0.618033988749895;
-        double h = ((double)rand() / (double)(RAND_MAX));
+        double h = (static_cast<double>(rand()) / static_cast<double>(RAND_MAX));
```
Comment (billbonney) [inline src/ui/AP2DataPlot2D.cpp]: typically you only need to cast one side to a double to force floating point division. if static cast look long winded for typing double(rand())/RAND_MAX is my preferred style.

File: src/ui/LogdataStorage.cpp [hunk 2]
```diff
@@ -221,15 +221,21 @@ double LogdataStorage::getTimeDivisor() const
     return m_timeDivisor;
 }
 
-QMap<QString, QStringList> LogdataStorage::getFmtValues() const
+QMap<QString, QStringList> LogdataStorage::getFmtValues(const bool filterStringValues) const
```
Comment (billbonney) [inline src/ui/LogdataStorage.cpp]: you are passing a copy of fundamental type, it doesn't need to be declared const

**Comment by billbonney** [comment]
i made some comments, which you can fix and then submit a new PR. I was kinda of playing with the new GitHub review tools..

And sorry for the delay in merging, I am just uber busy at the moment

File: src/ui/AP2DataPlot2D.cpp (hunk 1 above)
Comment (Arne-W) [inline src/ui/AP2DataPlot2D.cpp]: casting only one side seems to be enough but using double(vale) cast leads to a clang warning about "old style cast". So I would like to stay with the static_cast<>() style.

File: src/ui/LogdataStorage.cpp (hunk 2 above)
Comment (Arne-W) [inline src/ui/LogdataStorage.cpp]: You're right - I will fix it

File: src/ui/AP2DataPlot2D.cpp (hunk 1 above)
Comment (billbonney) [inline src/ui/AP2DataPlot2D.cpp]: interesting... c++ style casts are god for me 👍

**Pull request merged**